src/ConnectorInterface.php

```diff
@@ -4,5 +4,5 @@
 
 interface ConnectorInterface
 {
-    public function create($host, $port);
+    public function create($host, $port, array $contextOpts = []);
```

Comment:
I like the feature, but I'm not sure I like the API.

Let me quote from related [ticket #10](https://github.com/reactphp/socket-client/pull/10#issuecomment-109627103):

> IMO we should first figure out some use cases for this feature to see how this is actually going to be used.
>
> The socket-client component is quite lowlevel, so it's unlikely to be used by many users directly. It's probably more likely going to be used as part of some high level abstractions (say, a Redis or MySQL client implementation). These high level components depend on a `ConnectorInterface` and will likely use DI to inject this. They probably care little about *how* the connector works. Neither should they care about any "additional" parameters. They *do* care about the target (hostname and port).

Hence (IMHO) they should not have to take care of context options or connection timeouts.

See also the changes in the `SecureConnector`, most implementations of the `ConnectorInterface` will likely not care about the concept of "context options" and merely pass these through to their underlying `Connector`.

As an alternative, I would suggest passing context options to the constructor of the `Connector`.